eternalcombat-plugin/src/main/java/com/eternalcode/combat/border/BorderTriggerController.java

```diff
@@ -0,0 +1,62 @@
+package com.eternalcode.combat.border;
+
+import com.eternalcode.combat.fight.FightManager;
+import com.eternalcode.combat.fight.event.FightUntagEvent;
+import org.bukkit.Location;
+import org.bukkit.Server;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.bukkit.event.player.PlayerTeleportEvent;
+
+public class BorderTriggerController implements Listener {
+
+    private final BorderService borderService;
+    private final FightManager fightManager;
+    private final Server server;
+
+    public BorderTriggerController(BorderService borderService, FightManager fightManager, Server server) {
+        this.borderService = borderService;
+        this.fightManager = fightManager;
+        this.server = server;
+    }
+
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
+    void onMove(PlayerMoveEvent event) {
+        Location to = event.getTo();
+        Location from = event.getFrom();
+        if (to.getBlockX() == from.getBlockX() && to.getBlockY() == from.getBlockY() && to.getBlockZ() == from.getBlockZ()) {
+            return;
+        }
+
+        Player player = event.getPlayer();
+        if (!fightManager.isInCombat(player.getUniqueId())) {
+            return;
+        }
+
+        borderService.updateBorder(player, to);
```

Comment:
First, I would check if the player is in combat
```suggestion
    void onMove(PlayerMoveEvent event) {
        Player player = event.getPlayer();
        if (!fightManager.isInCombat(player.getUniqueId())) {
            return;
        }
            
        Location to = event.getTo();
        Location from = event.getFrom();
        if (to.getBlockX() == from.getBlockX() && to.getBlockY() == from.getBlockY() && to.getBlockZ() == from.getBlockZ()) {
            return;
        }

        borderService.updateBorder(player, to);
```

Reply:
```suggestion
this.borderService.updateBorder(player, to);
```

Reply:
```suggestion
    void onMove(PlayerMoveEvent event) {
        Location to = event.getTo();
        Location from = event.getFrom();
        if (to.getBlockX() == from.getBlockX() && to.getBlockY() == from.getBlockY() && to.getBlockZ() == from.getBlockZ()) {
            return;
        }

        Player player = event.getPlayer();
        if (!fightManager.isInCombat(player.getUniqueId())) {
            return;
        }

        this.borderService.updateBorder(player, to);
```

Reply:
odpytanie hashmapy może być bardziej kosztowne niż porównanie trzy razy intów w kontekście JIT - JVM optymalizuje sobie takie proste operacje, a hash mapa mimo, że jest faktycznie wydajna to pod spodem ma więcej obliczeń. Ale jestem otwarty na dyskusję
![image](https://github.com/user-attachments/assets/b05c3ad7-d8ec-4487-a1d3-532e09ade7f3)